libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt

```diff
@@ -320,55 +327,55 @@ class RustMatrixRoom(
     }
 
     override suspend fun retrySendMessage(transactionId: String): Result<Unit> =
-        withContext(coroutineDispatchers.io) {
+        withContext(roomDispatcher) {
             runCatching {
                 innerRoom.retrySend(transactionId)
             }
         }
 
     override suspend fun cancelSend(transactionId: String): Result<Unit> =
-        withContext(coroutineDispatchers.io) {
+        withContext(roomDispatcher) {
             runCatching {
                 innerRoom.cancelSend(transactionId)
             }
         }
 
     @OptIn(ExperimentalUnsignedTypes::class)
     override suspend fun updateAvatar(mimeType: String, data: ByteArray): Result<Unit> =
-        withContext(coroutineDispatchers.io) {
+        withContext(roomDispatcher) {
             runCatching {
                 innerRoom.uploadAvatar(mimeType, data.toUByteArray().toList())
             }
         }
 
     override suspend fun removeAvatar(): Result<Unit> =
-        withContext(coroutineDispatchers.io) {
+        withContext(roomDispatcher) {
             runCatching {
                 innerRoom.removeAvatar()
             }
         }
 
     override suspend fun setName(name: String): Result<Unit> =
-        withContext(coroutineDispatchers.io) {
+        withContext(roomDispatcher) {
             runCatching {
                 innerRoom.setName(name)
             }
         }
 
     override suspend fun setTopic(topic: String): Result<Unit> =
-        withContext(coroutineDispatchers.io) {
+        withContext(roomDispatcher) {
             runCatching {
                 innerRoom.setTopic(topic)
             }
         }
 
-    private suspend fun fetchMembers() = withContext(coroutineDispatchers.io) {
+    private suspend fun fetchMembers() = withContext(roomDispatcher) {
```

Comment:
You do not use `roomMembersDispatcher` here?

Reply:
We are not spamming it, so I don't think it's necessary, we are just calling it once.